iceoryx_utils/test/moduletests/test_unit_duration.cpp

```diff
@@ -18,218 +18,686 @@
 using namespace ::testing;
 using namespace iox::units;
 
-constexpr int KILO = 1000;
-constexpr int MEGA = 1000000;
-constexpr int GIGA = 1000000000;
-constexpr double tenTimesEpsilon = std::numeric_limits<double>::epsilon() * 10.;
+constexpr uint64_t KILO = 1000U;
+constexpr uint64_t MEGA = 1000000U;
+constexpr uint64_t GIGA = 1000000000U;
 
-TEST(Duration_test, convertDays)
+// BEGIN CONSTRUCTOR TESTS
+
+TEST(Duration_test, ConstructDurationWithLessNanosecondsThanOneSecond)
+{
+    constexpr uint64_t SECONDS{37U};
+    constexpr uint64_t NANOSECONDS{73U};
+    constexpr uint64_t EXPECTED_DURATION_IN_NANOSECONDS{SECONDS * GIGA + NANOSECONDS};
+
+    auto sut = Duration{SECONDS, NANOSECONDS};
+
+    EXPECT_THAT(sut.nanoSeconds<uint64_t>(), Eq(EXPECTED_DURATION_IN_NANOSECONDS));
+}
+
+TEST(Duration_test, ConstructDurationWithNanosecondsEqualToOneSecond)
+{
+    constexpr uint64_t SECONDS{13U};
+    constexpr uint64_t NANOSECONDS{GIGA};
+    constexpr uint64_t EXPECTED_DURATION_IN_NANOSECONDS{(SECONDS + 1U) * GIGA};
+
+    auto sut = Duration{SECONDS, NANOSECONDS};
+
+    EXPECT_THAT(sut.nanoSeconds<uint64_t>(), Eq(EXPECTED_DURATION_IN_NANOSECONDS));
+}
+
+TEST(Duration_test, ConstructDurationWithMoreNanosecondsThanOneSecond)
```

Comment:
I would add a test where you also construct  a duration object with 3 billion + x nanoseconds which should you then give you 3 seconds.
Additionally, when you think of:
- zero
- one
- many
- corner cases
- limits

You maybe should also construct a duration object with zero nanoseconds, both seconds and nanoseconds are zero, nanoseconds is `std::numeric_limits<uint32_t>::max` and where seconds and nanoseconds are both `std::numeric_limits<T>::max`